exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/SpanAdapter.java

```diff
@@ -101,11 +102,12 @@ private static InstrumentationLibrarySpans buildInstrumentationLibrarySpan(
 
   private static Map<Resource, Map<InstrumentationLibraryInfo, List<Span>>>
       groupByResourceAndLibrary(Collection<SpanData> spanDataList) {
-    Map<Resource, Map<InstrumentationLibraryInfo, List<Span>>> result = new HashMap<>();
+    IdentityHashMap<Resource, Map<InstrumentationLibraryInfo, List<Span>>> result =
```

Comment:
I understand that you changed variable type to IHM to explicitly state it's based on identity - a warning for future contributors, right?

Reply:
It is required by error prone - I guess because an IdentityHashMap technically doesn't follow the contract for Map.